src/subcommands/psa_export_public_key.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Exports a public key.
+
+pub use crate::cli::ParsecToolApp;
+use crate::error::ParsecToolError;
+use crate::subcommands::common::{OutputFileOpts, ProviderOpts};
+use crate::subcommands::ParsecToolSubcommand;
+use parsec_client::core::interface::requests::ProviderID;
+use parsec_client::core::operation_client::OperationClient;
+use parsec_interface::operations::psa_export_public_key;
+use parsec_interface::operations::{NativeOperation, NativeResult};
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::Write;
+use structopt::StructOpt;
+
+/// Exports a public key.
+#[derive(Debug, StructOpt)]
+#[structopt(name = "export-public-key")]
+pub struct PsaExportPublicKeySubcommand {
+    #[structopt(short = "k", long = "key-name")]
+    key_name: String,
+
+    #[structopt(flatten)]
+    provider_opts: ProviderOpts,
+
+    #[structopt(flatten)]
+    output_file_opts: OutputFileOpts,
+}
+
+impl TryFrom<&PsaExportPublicKeySubcommand> for NativeOperation {
```

Comment:
That's surely not related to this PR, but for my curiosity, why doing a `TryFrom` on a reference?

Reply:
Yes -- this might look strange at first glance.

We're doing `TryFrom` on a reference here because otherwise we would either lose ownership of the subcommand object (and therefore lose access to the variables it keeps, i.e. output file path, etc) _or_ we would need to create a copy/clone. This meant:

- we needed to be prepared to relinquish ownership of the subcommand object when converting it. I.e., we had to extract the variables we needed for the command from the struct _before_ conversion, which was a little messy.
- _or_, we needed to copy the struct. This would be okay if we could use `#[derive(Copy)]` -- but unfortunately, `String` doesn't implement `Copy` so we would have had to manually implement this for all of the subcommand objects.
- _or_, we needed to clone the struct. I seem to remember there being an issue with doing this, although I can quite recall. Either way, doing something like `self.clone()` seemed weird.

In an earlier version of `parsec-tool`, we implemented `TryFrom` on the object, not a reference. This caused awkward issues like being unable to have `String`-typed command-line arguments because they didn't implement `Copy`, etc.

In the end, conversion from a reference was far easier because the conversion really doesn't need ownership of the struct and it allows us to avoid the issues above.

Reply:
Ah what you say makes sense. That exposes the fact that, if there was not a reference, we would lose some information between the subcommand and the operation. I think that looks awkward because `TryFrom` should ideally be conversions between similar data but in different formats, with no loss of information and that's why you can give full ownership.
This perfectly fine as it is but maybe "hacking" on the semantics of the `TryFrom` trait. We could think about creating our own trait, like `ExtractOperation` that takes a reference and has the semantic of cloning the information inside the subcommand to create a new operation. But that's for the future 😅

Reply:
Agreed, it probably is hacking the semantics of the `TryFrom` trait. Would be happy to change this in the future.